### ISSUE TITLE
I propose a new overview section in the documentation

### ISSUE THREAD

**Issue by davidradl** [pytorch/pytorch#104623]
### 📚 The doc issue

I am new to PyTorch and what would have been useful to me was an overview of the project, at level higher than the code.
I think that this would be an 'and' not an 'or' with the existing docs and may help users who like this sort of conceptual view over the project.

I suggest we have a new Overview / concepts section in the documentation. That would include:

An overview of a typical PyTorch application, main and training data, model creation, loss and optimizer , training loop and test. I know these are also standard ML concepts, but I think it would help as they tie to specific PyTorch named concepts…
An overview of the PyTorch project in terms of python modules, models, C++ libraries, overview of the generation
An overview of the packaging and how the C++ and python libraries sit together.
API overviews.
Compiler overview contrasting the newer, older and recommended approaches.
GPU acceleration overview
Testing overview and approaches.
Any other overview perspective that might be useful
A starter Glossary for main terms to be defined and linked to as the term is used
The above should contain diagrams of the concepts and also link down to the existing doc. I hope that such an overview would make it more approachable to beginners and paint the big picture.

I am happy to put up initial content for it to be reviewed or work with others who are interested in this. I am hoping to draw lots of pictures!

All the sections could be built out over a series of prs.

**Comment by kiersten-stokes** [comment]
I would definitely be interested in helping you with this should this be given the go ahead!

**Comment by davidradl** [comment]
I assume the existing docs is geared towards the users of PyTorch. With one page on contributing. 

It seems to me that architecture would be useful to be documented for contributors. Maybe this could live initially in an overview. In time we could split the docs for the 2 main personas, PyTorch consumers and contributors.
 
@ezyang I found your post [http://blog.ezyang.com/2019/05/pytorch-internals/](http://blog.ezyang.com/2019/05/pytorch-internals/) very useful. My thinking is that a lot of this material (and other blogs you have done) could be brought into the main docs, completed and brought up to date. I could draw picture in svgs of the 'static' parts and separately the dynamic flows. Would you be ok with this? I am very happy to take direction around this, and am keen to bring it into the docs so it is more visible and useful.

I hope this overview / persona approach will make it easier for new members (like myself), who aspire to become strong contributors quickly.

**Comment by davidradl** [comment]
I see section https://github.com/pytorch/pytorch/blob/main/CONTRIBUTING.md#codebase-structure does show the code structure. I am thinking this could come up to a contributors section including an overview with a diagram, should these packages and dependancies / layers between them.

**Comment by ezyang** [comment]
sure. @msaroufim and @eellison recently added a bunch more docs too, they can also help with reviews

**Comment by albanD** [comment]
@subramen is actually working on building a new landing page for contributors, the current WIP version is at https://github.com/subramen/pytorch/wiki/Landing-matrix . Does that match what you have in mind?
Also I would suggest the core onboarding tutorials if you're looking for more advanced deep dives https://github.com/pytorch/pytorch/wiki/Core-Frontend-Onboarding

**Comment by subramen** [comment]
@davidradl great suggestions on integrating and updating this material. We've started working on assets to enable contributors like yourself ramp up quickly (https://github.com/subramen/pytorch/wiki/Landing-matrix) and your contributions would be very welcome!

**Comment by davidradl** [comment]
@subramen  @albanD yes this looks really good. Thanks for sharing. I come from an architectural background so like to see the big pictures in pictures. It seems that many in the community are very much lets get into the the code asap; which is great. I think that more of a conceptual view would help people like me and also forces the purpose and a module its dependancies, layers to be explicitly defined. This is also related to the generation.
